### PR TITLE
Update GraalVM to 22.3.1 JDK17

### LINKED ISSUE #6274: Upgrade to GraalVM based on JDK17
GraalVM 22.3.x was the last GraalVM that was based on JDK11. Future version of GraalVM - e.g. 23.x are going to be based on JDK17 and/or newer JDKs. 

Currently Enso uses **GraalVM 22.3.1 based on JDK11**. This complicates testing of new features and bugfixes done in `master` branch of http://github.com/oracle/graal repository as that requires switching between JDK11 & JDK17 and total recompilation of the code. Moreover certain features like JMH aren't ready for JDK17 - we have to update some libraries.

Let's switch to **GraalVM 22.3.x based on JDK17** as soon as "design partners" release is done.

Latest GraalVM update PR: https://github.com/enso-org/enso/pull/3663

### PR DESCRIPTION
**Pull request by Akirathan**
Fixes #6274

### Pull Request Description

Upgrade GraalVM to 22.3.2 based on JDK17.

**How to Upgrade**:
For `SDKMan` users:
```
sdk install java 22.3.1.r17-grl
sdk default java 22.3.1.r17-grl
```

For anybody else, just download the GraalVM distribution manually from https://github.com/graalvm/graalvm-ce-builds/releases/tag/vm-22.3.1 and set `JAVA_HOME` accordingly.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] Run benchmarks and ensure no regressions.
- [x] Try to install GraalPy and FastR and run interop tests
- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

**Comment by Akirathan** [comment]
I have tried:
- Building native image of `project-manager` and launcher, and running some Enso code.
- Yesterday's book club without any issues - AKKA appears to work on JDK17.
- Dry-run benchmarks on CI work and normal benchmarks work on my machine.

Failing tests on the CI:
- Some locale-related failure in `test/Table_Tests`: https://github.com/enso-org/enso/actions/runs/5015867538/jobs/8992003667?pr=6750#step:10:8284

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Update GraalVM to 22.3.2 JDK17 · enso-org/enso@226f9c7</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Comment by 4e6** [comment]
How is the benchmarks?

**Review by 4e6 — APPROVED** [review]
:tada:

**Comment by Akirathan** [comment]
> How is the benchmarks?

Do you think it is necessary to check benchmarks before integration? I guess that I should probably run them, at least from curiosity ..

**Comment by JaroslavTulach** [comment]
There are two groups of failing tests:
- [vectors of strings aren't equal](https://github.com/enso-org/enso/actions/runs/5015867538/jobs/8992003667?pr=6750#step:10:8285) - wow, [what a fix](https://github.com/enso-org/enso/pull/6750/commits/bae4271a0247fbf5bbf334aabb5b8a19a2814531) - it is _hard to spot a difference_!
- [version check problem](https://github.com/enso-org/enso/actions/runs/5015867538/jobs/8992003667?pr=6750#step:10:11426) - maybe we could remove the text after `Java-http-client/`...

**Comment by Akirathan** [comment]
Apparently, there are no `FastR`, neither `GraalPy` releases for `22.3.2`. I am reverting to `22.3.1` so that we can get interoperability with FastR and GraalPy.
- https://github.com/oracle/fastr/releases

<blockquote><div> GitHub</div><div><strong><a href="/oracle/fastr/releases">Releases · oracle/fastr</a></strong></div><div>A high-performance implementation of the R programming language, built on GraalVM. - oracle/fastr</div></blockquote>

**Comment by Akirathan** [comment]
Reverted to GraalVM JDK17 22.3.1 in 84a9e65df60ed66ef63dd40cbbb6bd91561ea736

**Comment by Akirathan** [comment]
Raised a question on GraalVM Slack polyglot channel "Why are FastR and GraalPy components not available on GraalVM JDK17 22.3.2" in https://graalvm.slack.com/archives/CP6RTC4LT/p1684762240737959

<blockquote><div><strong><a href="https://graalvm.slack.com/archives/CP6RTC4LT/p1684762240737959">Slack</a></strong></div></blockquote>

**Review by hubertp — APPROVED** [review]

**Comment by Akirathan** [comment]
[Package IDE (macos) job](https://github.com/enso-org/enso/actions/runs/5055177063/jobs/9072796578?pr=6750#step:10:2964) fails in signing the `.dmg` package with an output given [here](https://osxapps-ssl.itunes.apple.com/itunes-assets/Enigma126/v4/c3/6c/2d/c36c2db1-6324-fb7c-aa97-ab9297f2cde8/developer_log.json?accessKey=1685034536_5224328857340420911_udJSEDlY%2BxSzJJMyO%2FQa39BatgvJIztlidCeTp%2BO6h2eM35lvuGuodOV3O2taCqf%2BWjy9p%2FirowLuyBaNQjFS%2BMg1plz86revnu4Bbdkdcs8mlzlgHhqS7yRCdcPxGCaWgxFd0e4hZcndxODGMcdX4PpPSvLsCpkL3XebpUGKUk%3D)
```
The executable does not have the hardened runtime enabled.
```

**Comment by Akirathan** [comment]
Scheduled benchmarks in https://github.com/enso-org/enso/actions/runs/5056857474

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Benchmark Engine · enso-org/enso@5d07aa7</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Comment by Akirathan** [comment]
> Scheduled benchmarks in https://github.com/enso-org/enso/actions/runs/5056857474

Benchmark results are fine. I will get the MacOS signature working and merge this PR.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Benchmark Engine · enso-org/enso@5d07aa7</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Review by JaroslavTulach — APPROVED** [review]

**Review by somebody1234 — APPROVED** [review]
just reviewing the single line of TS code. it's too long (131 chars > 100), but it might be easier if i run the formatter on it

(edit: formatter has been run.)

**Comment by somebody1234** [comment]
weird, `prettier` is fine with that line

edit: oh, just forgot to pull

**Comment by Akirathan** [comment]
> weird, `prettier` is fine with that line
> 
> edit: oh, just forgot to pull

Thanks!

File: test/Table_Tests/src/In_Memory/Column_Format_Spec.enso [hunk 1]
```diff
@@ -307,7 +307,7 @@ spec_with_numeric_type name numeric_type =
 
         Test.specify "Column with locale" <|
             input = Column.from_vector "values" ["100000000", "2222", "3"] . parse numeric_type
-            expected = Column.from_vector "values" ["100 000 000,00", "2 222,00", "3,00"]
+            expected = Column.from_vector "values" ["100 000 000,00", "2 222,00", "3,00"]
```
Comment (Akirathan) [inline test/Table_Tests/src/In_Memory/Column_Format_Spec.enso:310]: This is a change of a Unicode symbol for a whitespace.

File: lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/components/GraalVMComponentConfiguration.scala [hunk 2]
```diff
@@ -14,10 +14,13 @@ class GraalVMComponentConfiguration extends RuntimeComponentConfiguration {
   ): Seq[GraalVMComponent] = {
     val optPythonComponent =
       if (os.hasPythonSupport) Seq(GraalVMComponent.python) else Seq()
+    val optRComponent =
+      if (os.hasRSupport) Seq(GraalVMComponent.R) else Seq()
```
Comment (Akirathan) [inline lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/components/GraalVMComponentConfiguration.scala:18]: This is a separate fix that I noticed recently from @xvcgreg's logs from another issue. Apparently, we try to install FastR on Windows, but that is not possible - it is not supported on Windows. Same case as with GraalPy, but FastR will, most probably, never be supported on Windows.

**Review by jdunkerley — APPROVED** [review]

**Review by jdunkerley — APPROVED** [review]
LGTM

File: test/Table_Tests/src/In_Memory/Column_Format_Spec.enso (hunk 1 above)
Comment (Akirathan) [reply test/Table_Tests/src/In_Memory/Column_Format_Spec.enso:310]: The old Unicode character was `U+A0` and the new one is `U+202F`

**Review by mwu-tow — APPROVED** [review]

**Pull request merged**